src/js/common/dataMap.js

```diff
@@ -1,54 +1,120 @@
+import RingBuffer from "./ringBuffer";
+
 let key = 0;
 const keygen = (reset = false) => {
   if (reset) key = 0;
   return key++;
 };
 
-export const DATA_KEYS = {
+// NOTE!!!!!!!!!!!!!!!
+// IF YOU CHANGE ANYTHING HERE; MAKE SURE IT GETS COPIED OVER TO BACKEND DATAKEYS.JS AS WELL!!!
+
+
+export const TYPES = {
+  INT8: 1,
+  INT16: 2,
+  FLOAT: 3,
+  BITFIELD: 4,
+  SPECIAL_ARRAY: 5, // 100 bytes
+  UINT8: 6,
+  UINT16: 7,
+  UINT32: 8,
+}
+
+/**
+ * @typedef {{ id: number,byteType: number }} DataMapEntry
+ * Data Keys are a hash of DataMapEntry objects
+ * @typedef {Object.<string, DataMapEntry>} DataKeys
+ */
+
+ export const DATA_MAP = {
```

Comment:
when you add a new key in the backend version, just copy and paste that entire DATA_MAP and replace this one.

Reply:
Awesome, much cleaner :) Thanks!